src/control/server/mgmt_system.go

```diff
@@ -663,18 +663,23 @@ func (svc *mgmtSvc) SystemStop(ctx context.Context, req *mgmtpb.SystemStopReq) (
 	}
 	var fResp *fanoutResponse
 
-	fReq.Method = control.PrepShutdownRanks
-	// if not forced, update membership on rank error
-	fResp, _, err = svc.rpcFanout(ctx, fReq, !req.Force)
-	if err != nil {
-		return
-	}
-	if !fReq.Force && fResp.Results.Errors() != nil {
-		// return early if not forced and prep shutdown fails
-		resp, err = processStopResp("prep shutdown", fResp, svc.events)
-		return
+	if !req.Force {
```

Comment:
The thing that i am not clear about (and not related to this patch) is that one can also shutdown a specific list of ranks. The prep shutdown makes sense only when shutdown all the ranks IMO. Otherwise, you still want exclusion to happen. Any thoughts?

Reply:
yes that makes sense, the prep shutdown implementation initially was generic and not specifically related to exclusion (it's just a mechanism to call into the prep shutdown drpc handler)